guide/src/conversions.md

```diff
@@ -1,8 +1,103 @@
 # Type Conversions
 
+In this portion of the guide we'll talk about the mapping of Python types to Rust types offered by PyO3, as well as the traits available to perform conversions between them.
+
+## Mapping of Rust types to Python types
+
+When writing functions callable from Python (such as `#[pyfunction]`s and `#[pymethods]`) consult the tables below to determine the correct Rust types to use for argument and return types.
+
+### Argument Types
+
+When you accept a function argument, you can choose to either use Rust library types or PyO3's Python-native types. (See the next section for discussion on when to use each.)
+
+The table below contains the Python type and the corresponding Rust library type or PyO3 Python-native type that can be used as a function argument:
+
+| Python        | Rust                            | Rust (Python-native) |
+| ------------- |:-------------------------------:|:--------------------:|
+| `Any`         | -                               | `&PyAny`             |
+| `str`         | `String`, `&str`, `Cow<str>`    | `&PyUnicode`         |
+| `bytes`       | `&[u8]`                         | `&PyBytes`           |
+| `bool`        | `bool`                          | `&PyBool`            |
+| `int`         | Any integer type (`i32`, `u32`, `usize`, etc) | `&PyLong` |
```

Comment:
Should we mark cases where the conversion can fail, like here?

Reply:
Not sure. Technically all conversion can fail with a `TypeError` ;)

Reply:
If you don't have a strong opinion on this, I think for the moment let's keep the doc simpler.